src/include/migraphx/program.hpp

```diff
@@ -89,10 +88,21 @@ struct program
     friend bool operator==(const program& x, const program& y);
     friend bool operator!=(const program& x, const program& y) { return !(x == y); }
 
+    // module related api
+    module* create_module(const std::string& name);
+    module* get_module(const std::string& name);
+    const module* get_module(const std::string& name) const;
+    void remove_module(const std::string& name);
+
     module* get_main_module();
     const module* get_main_module() const;
 
+    std::vector<const module*> get_modules() const;
+
     private:
+    void mod_from_val(module_ref mod, const value& v,
```

Comment:
This could be declared static and moved to the cpp file.

Reply:
fixed